### PR TITLE
Improve testing section in Pulp contributor docs

### LINKED ISSUE #7475: otel_metrics task is too expensive — high DB CPU usage and long execution time
The otel_metrics scheduled task (`pulpcore.app.tasks.telemetry.otel_metrics`) runs every 5 minutes and currently takes approximately 1 minute and 30 seconds to complete. During execution, the underlying query consumes ~3 vCPUs from the database.

Here is where we believe the issue is happening (pulpcore/app/tasks/telemetry.py:31-33):
```
  space_usage_per_domain = Artifact.objects.values("pulp_domain__name").annotate(
      total_size=Sum("size", default=0)
  )
```
- A telemetry task that is meant to be lightweight is placing significant load on the database.
- 3 vCPU consumption for a single periodic query reduces capacity available for actual content operations (sync, publish, etc.).
- The task takes 1m30s out of every 5-minute cycle (30% duty cycle), meaning a worker is occupied with telemetry for a disproportionate amount of time.

### PR DESCRIPTION
**Pull request by mdellweg**
Mentioned `prestart`, `pminio`, `pfixtures` and `phelp`.

fixes #7475
https://pulp.plan.io/issues/7475

### CODE REVIEW

File: CHANGES/7475.doc [hunk 1]
```diff
@@ -0,0 +1,2 @@
+Improved testing section in Pulp contributor docs.
```
Comment (ipanova) [inline CHANGES/7475.doc:1]: we generally try to keep changes as one liner

File: docs/contributing/tests.rst [hunk 2]
```diff
@@ -60,6 +60,21 @@ Functional tests require a valid *pulp-smash*
...
+Using pulplift
+^^^^^^^^^^^^^^
+
+When running one of the `pulp3-source-*` boxes in `pulplift`, the smash config is already in place.
+Also, all the services are running.
+They should be restarted with `prestart` if any pulp code (not test code) has been changed.
+
+When testing S3 support, you can start and configure a local `minio` container with `pminio`.
+
+Pulp functional tests use a set of upstream fixture repositories.
```
Comment (ipanova) [inline docs/contributing/tests.rst:72]: how about providing the link to the fixtures?

**Review by ipanova — APPROVED** [review]

File: docs/contributing/tests.rst [hunk 3]
```diff
@@ -60,6 +60,22 @@ Functional tests require a valid *pulp-smash*
 `config <https://pulp-smash.readthedocs.io/en/latest/configuration.html>`_ file.
 This can be created with `pulp-smash settings create`.
 
+Using pulplift
+^^^^^^^^^^^^^^
+
+When running one of the `pulp3-source-*` boxes in `pulplift`, the smash config is already in place.
+Also, all the services are running.
```
Comment (bmbouter) [inline docs/contributing/tests.rst:67]: The newline here makes the unrendered docs for difficult to read (to me). I put the newline style guide questions on the pulpcore agenda for the next meeting. I want us to agree one way or the other before our docs slows become a mixture of styles.

**Review by bmbouter — APPROVED** [review]
This looks great, thank you!

**Pull request merged**